### DISCUSSION TITLE
How does this differ from the DataDog setup?

### DISCUSSION THREAD

**Discussion by benpettman** [odigos-io/odigos#68] [Q&A] [answered]
Hello, great looking solution. I currently have some K8 clusters with the datadog agent installed and configured to collect traces for these along with other metrics. 

How does odigos differ or benefit more than the DD native integration?

**Comment by edeNFed** [comment, marked as answer]
Here a few differences between Datadog agent and Odigos:

- Automatic instrumentation by default: Odigos automatically instruments any newly-deployed application by recognizing the programming language and applying the relevant instrumentation
- Support for Go auto instrumentation
- Vendor neutrality: Odigos produces tracing data in OpenTelemetry format which is currently supported by all the popular Observability backends and allow switching vendors with a click of a button
- Collectors management: Odigos automatically scale collectors according to the amount of observability traffic to handle spikes in traffic.

I think that the bottom line is higher quality data without vendor lock in.

**Reply by benpettman** [reply]
Thanks for replying. But in my setup I install Datadog as a helm chart, put a couple of env vars on my deployment and then everything is traced. But I guess because I have installed the agent inside the container. 

Are you saying I don’t need this part?

**Reply by edeNFed** [reply]
Yes exactly. You don't need to do any code changes / adding env vars. Everything happens automatically.

**Reply by benpettman** [reply]
I will give it a try

**Reply by nashetali837** [reply]
We dont need code changes/adding env vars. Great Tool...
